code/datums/components/chem_booster.dm

```diff
@@ -0,0 +1,237 @@
+	var/boost_on = FALSE
+	///Stores the value with which effect_mult is modified
+	var/boost_amount = 0
+	///Normal boost
+	var/boost_tier1 = 1
+	///Overcharged boost
+	var/boost_tier2 = 3
+	///Boost icon, image cycles between 2 states
+	var/boost_icon = "cboost_t1"
+	///Item connected to the system
+	var/obj/item/connected_weapon
+	///When was the effect activated. Used to activate negative effects after a certain amount of use
+	var/processing_start
+	COOLDOWN_DECLARE(chemboost_activation_cooldown)
+
+/datum/component/chem_booster/Initialize()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+	update_boost(boost_tier1)
+
+/datum/component/chem_booster/Destroy(force, silent)
+	QDEL_NULL(configure_action)
+	wearer = null
+	return ..()
+
+/datum/component/chem_booster/RegisterWithParent()
+	. = ..()
+	configure_action = new(parent)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(parent, list(COMSIG_ITEM_EQUIPPED_NOT_IN_SLOT, COMSIG_ITEM_DROPPED), .proc/dropped)
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED_TO_SLOT, .proc/equipped)
+	RegisterSignal(configure_action, COMSIG_ACTION_TRIGGER, .proc/configure)
+
+/datum/component/suit_autodoc/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, list(
+		COMSIG_PARENT_EXAMINE,
+		COMSIG_ITEM_EQUIPPED_NOT_IN_SLOT,
+		COMSIG_ITEM_DROPPED,
+		COMSIG_ITEM_EQUIPPED_TO_SLOT))
+	QDEL_NULL(configure_action)
+
+/datum/component/chem_booster/proc/examine(datum/source, mob/user)
+	SIGNAL_HANDLER
+	to_chat(user, "<span class='notice'>The chemical system currently holds [resource_storage_current]u of green blood. Its' enhancement level is set to [boost_amount].</span>")
+
+/datum/component/chem_booster/proc/dropped(datum/source, mob/user)
+	SIGNAL_HANDLER
+	if(!iscarbon(user))
+		return
+	if(boost_on)
+		on_off()
+	manage_weapon_connection()
+
+	configure_action.remove_action(wearer)
+	wearer = null
+
+/datum/component/chem_booster/proc/equipped(datum/source, mob/equipper, slot)
+	SIGNAL_HANDLER
+	if(!isliving(equipper))
+		return
+	wearer = equipper
+	configure_action.give_action(wearer)
+
+/datum/component/chem_booster/process()
+	if(resource_storage_current < resource_drain_amount)
+		to_chat(wearer, "<span class='warning'>Insufficient resources to maintain operation.</span>")
+		on_off()
+	resource_storage_current = max(resource_storage_current - resource_drain_amount, 0)
+
+	if(world.time - processing_start > 12 SECONDS)
+		wearer.overlay_fullscreen("degeneration", /obj/screen/fullscreen/infection, 1)
+
+/**
+ *	Opens the radial menu with everything
+ */
+/datum/component/chem_booster/proc/configure(datum/source)
+	var/mob/living/carbon/human/H = wearer
+
+	var/list/radial_options = list(
+		"on_off" = image(icon = 'icons/mob/radial.dmi', icon_state = "cboost_on_off"),
+		"connect" = image(icon = 'icons/mob/radial.dmi', icon_state = "cboost_connect"),
+		"boost" = image(icon = 'icons/mob/radial.dmi', icon_state = "[boost_icon]"),
+		)
```

Comment:
Is there a reason this isnt a static list?

Reply:
changed